lib/test.js

```diff
@@ -457,6 +462,26 @@ class Test {
 		this.timeoutTimer = null;
 	}
 
+	teardown(callback) {
+		this.teardowns.push(callback);
+	}
+
+	async runTeardowns() {
+		const errors = [];
+		for (const teardown of this.teardowns) {
+			try {
+				await teardown(); // eslint-disable-line no-await-in-loop
+			} catch (error) {
+				errors.push(error);
```

Comment:
(I don't know how GitHub will end up ordering these comments, so you may have to read ahead for this to make sense…)

Instead of returning all errors below, let's use `saveFirstError()`.

Reply:
Fair enough. Makes sense when reading ahead.